pkg/apis/networking/validation/validation.go

```diff
@@ -597,3 +598,78 @@ func allowInvalidWildcardHostRule(oldIngress *networking.Ingress) bool {
 	}
 	return false
 }
+
+// ValidateClusterCIDRConfigName validates that the given name can be used as an
+// ClusterCIDRConfig name.
+var ValidateClusterCIDRConfigName = apimachineryvalidation.NameIsDNSLabel
+
+// ValidateClusterCIDRConfig validates a clusterCIDRConfig.
+func ValidateClusterCIDRConfig(ccc *networking.ClusterCIDRConfig) field.ErrorList {
+	allErrs := apivalidation.ValidateObjectMeta(&ccc.ObjectMeta, false, ValidateClusterCIDRConfigName, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidateClusterCIDRConfigSpec(&ccc.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+// ValidateClusterCIDRConfigSpec validates clusterCIDRConfig Spec.
+func ValidateClusterCIDRConfigSpec(spec *networking.ClusterCIDRConfigSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, unversionedvalidation.ValidateLabelSelector(spec.NodeSelector, fldPath.Child("nodeSelector"))...)
+
+	maxIPv4PerNodeMaskSize := int32(32)
+	maxIPv6PerNodeMaskSize := int32(128)
+
+	hasIPv4 := spec.IPv4 != nil
+	hasIPv6 := spec.IPv6 != nil
+
+	// Validate if CIDR is configured for at least one IP Family(IPv4/IPv6).
+	if !hasIPv4 && !hasIPv6 {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("ipv4", "ipv6"), *spec, "either IPv4 or IPv6 CIDR must be configured, both cannot be empty"))
+	}
+
+	// Validate configured IPv4 CIDR and PerNodeMaskSize.
+	if hasIPv4 {
+		if !netutils.IsIPv4CIDRString(spec.IPv4.CIDR) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("ipv4", "cidr"), *spec.IPv4, "provided CIDR is not a valid IPv4 CIDR"))
+		}
+		if spec.IPv4.PerNodeMaskSize < 0 || spec.IPv4.PerNodeMaskSize > maxIPv4PerNodeMaskSize {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("ipv4", "perNodeMaskSize"), *spec.IPv4, "provided IPv4 PerNodeMaskSize is invalid, Valid values: [0,32]"))
+		}
+	}
+
+	// Validate configured IPv6 CIDR and PerNodeMaskSize.
+	if hasIPv6 {
+		if !netutils.IsIPv6CIDRString(spec.IPv6.CIDR) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("ipv6", "cidr"), *spec.IPv6, "provided CIDR is not a valid IPv6 CIDR"))
+		}
+		if spec.IPv6.PerNodeMaskSize < 0 || spec.IPv6.PerNodeMaskSize > maxIPv6PerNodeMaskSize {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("ipv6", "perNodeMaskSize"), *spec.IPv6, "provided IPv6 PerNodeMaskSize is invalid, Valid values: [0,128]"))
+		}
+	}
+
+	// Validate PerNodeMaskSize if both IPv4 and IPv6 CIDRs are configured.
+	// 32 - spec.IPv4.PerNodeMaskSize == 128 - spec.IPv6.PerNodeMaskSize must be satisfied.
```

Comment:
I expect networking people screaming here :smile: 

we should document it well

Reply:
Added more comments here, @rahulkjoshi Is there any other reason that I might have missed for this check

Reply:
🙃 yeah this is our initial compromise to make sure we don't need to change the scheduler or be too sophisticated. Otherwise, the scheduler needs to understand which IPfamilies a pod wants, which ones a node has, and how many free IPs in each family the node has. It's just easier to enforce sameness for now, until we decide where we want to take this.

But yeah, there will be some people who are un-amused here...

Reply:
my experience is that is easier to work with masks, they are simpler to think about ... and you spot errors immediately

```
ipv4:
  cidr: 10.0.0.0/16
  perNodeMaskSize: 15
```

vs 
```
perNodeSize: 65537
ipv4CIDR: 10.0.0.0/16
```

or 
```
perNodeBits: 15
ipv4CIDR: 10.0.0.0/16
```

Reply:
though, if we set this common field , we are giving up in being able to decouple mask sizes in the future, are we sure we'll always use the same mask for both families? since this is alpha I think we can go with this restriction now and reevaluate later with more feedback